src/pybind/mgr/mgr_module.py

```diff
@@ -330,6 +330,70 @@ def device_class_counts(self) -> Dict[str, int]:
 HandlerFuncType = Callable[..., Tuple[int, str, str]]
 
 
+def _extract_target_func(
+    f: HandlerFuncType
+) -> Tuple[HandlerFuncType, Dict[str, Any]]:
+    """In order to interoperate with other decorated functions,
+    we need to find the original function which will provide
+    the main set of arguments. While we descend through the
+    stack of wrapped functions, gather additional arguments
+    the decorators may want to provide.
+    """
+    wrapped = getattr(f, "__wrapped__", None)
+    if not wrapped:
+        return f, {}
```

Comment:
I'd suggest to always avoid `getattr` when the attribute is a literal/known at 'build time', and use instead the [EAFP approach](https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Exceptions):
```suggestion
    try:
        wrapped = f.__wrapped__
    except AttributeError:
        return f, {}
```

Reply:
Sure.